AG-15064 - Add snyk ignore script + sync up shared security workflow

taktran opened the pull request.

taktran committed:
git subrepo pull external/ag-shared

subrepo:
  subdir:   "external/ag-shared"
  merged:   "5155d73341"
upstream:
  origin:   "https://github.com/ag-grid/ag-shared.git"
  branch:   "latest"
  commit:   "fb19c1121c"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "30db3b8"

taktran committed:
AG-15064 - Make calculate-pkg-shards shareable between grid/charts

taktran committed:
AG-15064 - Sync up snyk security workflow with grid

taktran committed:
AG-15064 - Add shared snyk ignore script to nx task

taktran commented on .github/workflows/security.yml:
```diff
@@ -32,8 +43,120 @@ jobs:
+                  CURRENT_COMMIT_SHA: ${{ github.sha }}
+                  IS_SUCCESS: ${{ needs.code-dependency-scan.result != 'failure' && needs.sast-code-scan.result != 'failure' }}
+
+            # - name: Create Jira Ticket
```
Will work this out in another pass

taktran committed:
AG-15064 - Adapt security workflow for output folder

taktran committed:
AG-15064 - Allow notification title to be different based on library

taktran committed:
AG-15064 - Use ag bot script for sending slack message

taktran commented on external/ag-shared/github/actions/slack-integration/action.yml:
```diff
@@ -0,0 +1,75 @@
+              JOB_ID: ${{ github.run_id }}
+              REPO_NAME: ${{ github.repository }}
+
+        - name: Convert ctrf output to Slack message blocks
```
I want to merge this and the send slack message step in future

sdwvit commented on external/ag-shared/github/actions/get-last-failed-step/action.yml:
```diff
@@ -0,0 +1,47 @@
+name: get-last-failed-step
+author: sdwvit
```
;)

sdwvit replied:
```suggestion
```

sdwvit commented:
there is some code sharing opportunity. otherwise lgtm.

please test before merge

taktran committed:
git subrepo push external/ag-shared

subrepo:
  subdir:   "external/ag-shared"
  merged:   "ddd554b25d"
upstream:
  origin:   "https://github.com/ag-grid/ag-shared.git"
  branch:   "latest"
  commit:   "ddd554b25d"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "30db3b8"
